language/sygus.tex

```diff
@@ -3015,6 +3013,77 @@ \section{Examples}%
 \end{lstlisting}
 \end{example}
 
+
+\begin{example}[PBE with Oracle Function Symbols]
+Oracle function symbols are used to represent black-box functions within specifications. Consider a PBE problem where the user has an external oracle and wishes to synthesise a summary of the external oracle for a specific set of inputs. This can be encoded
+using an oracle function symbol (note this is syntactic sugar for an oracle assumption and an uninterpreted function):
+\begin{lstlisting}[language=SyGuS]
+(set-logic BV)
+(set-feature :oracles true) 
+(synth-fun f ((x (_ BitVec 64))) (_ BitVec 64))
+(declare-oracle-fun target ((_ BitVec 64)) (_ BitVec 64) binaryname)
+
+(constraint (= (f #x28085a970e13e12c) (target #x28085a970e13e12d)))
+(constraint (= (f #xbe5341bebd2a0749) (target #xbe5341bebd2a0749)))
+(constraint (= (f #xe239460eed2cc34e) (target #xe239460eed2cc34f)))
+(constraint (= (f #xac5b1b5e9b236b10) (target #xac5b1b5e9b236b11)))
+(constraint (= (f #x4069a4c7173e1786) (target #x4069a4c7173e1786)))
+(constraint (= (f #x39419062091119a6) (target #x39419062091119a6)))
+(constraint (= (f #x49aeeca628644ee0) (target #x49aeeca628644ee0)))
+(constraint (= (f #x75e5bc2a07c77c97) (target #x75e5bc2a07c77c97)))
+(constraint (= (f #x4c5ee4be98c5ee7d) (target #x4c5ee4be98c5ee7d)))
+(constraint (= (f #xcd67bd5beaac575e) (target #xcd67bd5beaac575e)))
+(check-synth)
+\end{lstlisting}
+In this way a PBE problem can be encoded without the user knowing the correct behaviour of the target function explicitly.
```

Comment:
Can you have a sentence that mentions `binaryname` and practically what this means for the solver.

Reply:
```suggestion
In this way, a PBE problem can be encoded without the user knowing the correct behaviour of the target function explicitly. 
Instead, the target function is implemented by the oracle \lstinline{binaryname}, 
and the solver can call \lstinline{binaryname} with an input given in a constraint 
and \lstinline{binaryname} returns the result of applying \lstinline{target} to the input.
```